src/Sentry/Internal/Hub.cs

```diff
@@ -62,7 +62,7 @@ internal class Hub : IHubEx, IDisposable
         _enricher = new Enricher(options);
 
         // An integration _can_ deregister itself, so make a copy of the list before iterating.
-        var integrations = options.Integrations?.ToList() ?? Enumerable.Empty<ISdkIntegration>();
+        var integrations = options.Integrations.ToList();
```

Comment:
Could we get around iterating on `Integrations` below without allocating the list?

Reply:
My first draft didn't have the `ToList()` and some test was failing. One integration was removing another integration.

It's probably this one (which will work fine now with the enum flags):
```csharp
internal class SentryDiagnosticListenerIntegration : ISdkIntegration
{
    public void Register(IHub hub, SentryOptions options)
    {
        if (!options.IsPerformanceMonitoringEnabled)
        {
            options.Log(SentryLevel.Info, "DiagnosticSource Integration is disabled because tracing is disabled.");
            options.DisableDiagnosticSourceIntegration();
            return;
        }

        var subscriber = new SentryDiagnosticSubscriber(hub, options);
        DiagnosticListener.AllListeners.Subscribe(subscriber);
    }
}
```

Reply:
In fact no it won't be working, there is a duality with `SentryDiagnosticListenerIntegration` as it's internaly provided or provided by `Sentry.DiagnosticSource`. 

I feel like it's useless to remove itself from the options. I'm removing the `ToList`, I just adapted a test to check the log instead.